libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -635,6 +680,45 @@ void CodeTransform::visitStatements(vector<Statement> const& _statements)
 	for (auto const& statement: _statements)
 	{
 		freeUnusedVariables();
+
+		if (!m_delayedReturnVariables.empty() && (
+			holds_alternative<VariableDeclaration>(statement) ||
+			holds_alternative<Leave>(statement) ||
+			holds_alternative<ForLoop>(statement) ||
+			holds_alternative<Block>(statement) ||
+			holds_alternative<Switch>(statement) ||
+			holds_alternative<If>(statement) ||
+			[&](){
+				ReferencesCounter referencesCounter{ReferencesCounter::CountWhat::OnlyVariables};
```

Comment:
I don't think detects assignments to return variables.

Reply:
Ah sorry, it does. It just does not detect the names of variable declarations.

Reply:
I checked - it does. Maybe it *shouldn't*, but it does :-). ``ReferencesCounter`` only overrides the identifier visit and ``ASTWalker`` visits assignment LHSs. (If it didn't detect assignments, I'm also pretty sure this would utterly break tests.)